Helper/Order.php

```diff
@@ -1094,8 +1094,12 @@ public function processExistingOrder($quote, $transaction)
     public function processNewOrder($quote, $transaction)
     {
         /** @var OrderModel $order */
-        $order = $this->quoteManagement->submit($quote);
-        if ($order === null) {
+        $this->quoteManagement->submit($quote);
+        // When the create_order hook (thread 1) takes a lot of time to execute and returns a timeout, the authorize/payment hook (thread 2) is sent to Magento.
+        // It updates the order prior to the create_order hook (thread 1) process.
+        // This ensures that the returned order in the create_order hook is the latest updated order.
+        $order = $this->getExistingOrder($quote->getReservedOrderId());
```

Comment:
Very soon (like today) we will stop to use reserverOrderId. Can we use parentQuoteID instead?

Reply:
@vitaliyreznikov I've updated the code to get the order by the quote id

Reply:
Can you re-check?

Reply:
I am sorry John I didn't consider that function getOrderByQuoteId is a litle heavy. Could you please make one more change:
- At line 1097 save order_id
- Find order by id and only if doesn't exist (3p plugin changed number) find by quote id. I think we have a helper for this.

Reply:
@vitaliyreznikov 
**"Find order by id and only if doesn't exist (3p plugin changed number) find by quote id. I think we have a helper for this."**
I'm confused about this point. The order id is the primary key and it can't be changed by the 3rd party plugin. Are you saying that we should fetch the order by the increment id first and then if it doesn't exist, fetch it by the quote id as a fallback?

Reply:
Thank's for your explanation, John. You are right. Could we use order id for our purpose?

Reply:
@vitaliyreznikov I've updated the code. Could you please re-check?